core_backend/add_new_data_to_db.py

```diff
@@ -0,0 +1,412 @@
+    content = search_results[str(random.randint(0, 3))]
+
+    # Get content text and use to generate feedback text using LLMs
+    content_text = content["title"] + " " + content["text"]
+    generated_text = generate_feedback(query_text, content_text, feedback_sentiment)
+
+    feedback_text = (
+        generated_text["output"]
+        if generate_feedback_text and generated_text is not None
+        else ""
+    )
+
+    data = {
+        "query_id": query_id,
+        "session_id": 0,
+        "content_id": content["id"],
+        "feedback_sentiment": feedback_sentiment,
+        "feedback_text": feedback_text,
+        "feedback_secret_key": feedback_secret_key,
+    }
+    response = (
+        save_single_row(endpoint, data) if not pd.isna(feedback_sentiment) else None
+    )
+    if response is not None:
+        return (query_id,)
+
+    return None
+
+
+def process_urgency_detection(_id: int, text: str) -> tuple | None:
+    """
+    Process and add urgency detection to DB
+    """
+    endpoint = f"{HOST}/urgency-detect"
+    data = {
+        "message_text": text,
+    }
+
+    response = save_single_row(endpoint, data)
+    if "is_urgent" in response:
+        return (response["is_urgent"],)
+    return None
+
+
+def create_random_datetime_from_string(date_string: str) -> datetime:
+    """
+    Create a random datetime from a date in the format "%d-%m-%y
+    to today
+    """
+    date_format = "%d-%m-%y"
+
+    start_date = datetime.strptime(date_string, date_format)
+
+    time_difference = datetime.now() - start_date
+    random_number_of_days = random.randint(0, time_difference.days)
+
+    random_number_of_seconds = random.randint(0, 86399)  # Number of seconds in one day
+
+    random_datetime = start_date + timedelta(
+        days=random_number_of_days, seconds=random_number_of_seconds
+    )
+    return random_datetime
+
+
+def update_date_of_records(models: list, random_dates: list, api_key: str) -> None:
+    """
+    Update the date of the records in the database
+    """
+    session = next(get_session())
+    hashed_token = get_key_hash(api_key)
+    user = session.execute(
+        select(UserDB).where(UserDB.hashed_api_key == hashed_token)
+    ).scalar_one()
+    for model in models:
+        session = next(get_session())
+        queries = [
+            c for c in session.query(model[0]).all() if c.user_id == user.user_id
+        ]
+
+        if len(queries) > len(random_dates):
+            random_dates = random_dates + [
+                create_random_datetime_from_string(start_date)
+                for _ in range(len(queries) - len(random_dates))
+            ]
+        for query, date in zip(queries, random_dates):
+            # Set the date attribute to the random date
+            setattr(query, model[1], date)
+            session.merge(query)
+        session.commit()
+
+
+if __name__ == "__main__":
+    HOST = auth_url = f"http://{args.domain}"
```

Comment:
I was using docker compose as suggested by the test steps.
Does this work with Caddy also? Can we just accept the AAQ URL instead of domain so that user can define whether it's HTTP or HTTPS?

Reply:
Adding the url instead makes sense. But the issue is this script has to be run inside the maching hosting AAQ, because:
- We are accessing the DB
- We are using The deployed LiteLLM. 
So we will endup using either localhost:8000 or localhost/api. Nothing else